File: src/main/java/wraith/fabricaeexnihilo/util/VoxelShapeHelper.java
package wraith.fabricaeexnihilo.util;

import net.minecraft.util.function.BooleanBiFunction;
import net.minecraft.util.shape.VoxelShape;
import net.minecraft.util.shape.VoxelShapes;

public final class VoxelShapeHelper {

    private VoxelShapeHelper() {}

    public static VoxelShape union(VoxelShape... shapes) {
        var output = VoxelShapes.empty();
        for(var shape : shapes){
            output = VoxelShapes.combine(output, shape, BooleanBiFunction.OR);
        }
        return output.simplify();
    }

    public static VoxelShape intersection(VoxelShape... shapes) {
        var output = VoxelShapes.fullCube();
        for(var shape : shapes){
            output = VoxelShapes.combine(output, shape, BooleanBiFunction.AND);
        }
        return output.simplify();
    }

}

File: src/main/kotlin/exnihilofabrico/util/VoxelShapeHelper.kt
package exnihilofabrico.util

import net.minecraft.util.BooleanBiFunction
import net.minecraft.util.shape.VoxelShape
import net.minecraft.util.shape.VoxelShapes

object VoxelShapeHelper {
    fun union(vararg shapes: VoxelShape): VoxelShape {
        var output = VoxelShapes.empty()
        for(shape in shapes){
            output = VoxelShapes.combine(output, shape, BooleanBiFunction.OR)
        }
        return output.simplify()
    }

    fun intersection(vararg shapes: VoxelShape): VoxelShape {
        var output = VoxelShapes.fullCube()
        for(shape in shapes){
            output = VoxelShapes.combine(output, shape, BooleanBiFunction.AND)
        }
        return output.simplify()
    }
}